agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AzureMonitorSpanProcessor.java

```diff
@@ -24,13 +26,22 @@ public void onStart(Context parentContext, ReadWriteSpan span) {
     // first before creating child span
 
     Span parentSpan = Span.fromContextOrNull(parentContext);
+    // Azure function host is emitting request, java agent doesn't.
+    // parentSpan is not an instanceof ReadableSpan here, thus need to update operationName before
+    // checking for ReadableSpan
+    if (ConfigurationBuilder.inAzureFunctionsWorker()) {
+      AzureFunctionsCustomDimensions customDimensions =
+          AzureFunctionsCustomDimensions.fromContext(parentContext);
+      if (customDimensions != null && customDimensions.operationName != null) {
+        span.setAttribute(AiSemanticAttributes.OPERATION_NAME, customDimensions.operationName);
+      }
+    }
     if (!(parentSpan instanceof ReadableSpan)) {
```

Comment:
(just a formatting suggestion since don't need the parentSpan for the Az Fn part)
```suggestion
    // Azure function host is emitting request, java agent doesn't.
    // parentSpan is not an instanceof ReadableSpan here, thus need to update operationName before
    // checking for ReadableSpan
    if (ConfigurationBuilder.inAzureFunctionsWorker()) {
      AzureFunctionsCustomDimensions customDimensions =
          AzureFunctionsCustomDimensions.fromContext(parentContext);
      if (customDimensions != null && customDimensions.operationName != null) {
        span.setAttribute(AiSemanticAttributes.OPERATION_NAME, customDimensions.operationName);
      }
    }

    Span parentSpan = Span.fromContextOrNull(parentContext);
    if (!(parentSpan instanceof ReadableSpan)) {
```